### DISCUSSION TITLE
Rotating a model to point at a Geopoint

### DISCUSSION THREAD

**Discussion by Mattc2021** [pelicanmapping/osgearth#2284] [Support Q&A]
Hello everyone, I was curious if it was possible to rotate a model that has been loaded onto OSGEarth to point to a specific Geopoint. The main issue I'm having is getting the actual model to rotate. So let's assume that the Model has been loaded and placed at a specific Geopoint. I have 

```
auto GroupNode = Group->getChild(0);
Group->removeChild(0);
osg::MatrixTransform* transform = new osg::MatrixTransform;
const double angle = 0.8;
const osg::Vec3d axis(0, 0, 1);
transform->setMatrix(osg::Matrix::rotate(angle, axis));
transform->setName("Transform");
transform->addChild(GroupNode);
Group->insertChild(0, transform);
```

which should allow me to adjust the angle. I read that I would potentially have to remove the child from the group in order to handle the traversals correctly, but was curious if what I'm doing is possible.

**Comment by gwaldron** [comment]
How are you incorporating the GeoPoint? Are you using a GeoTransform?

Be careful with that `auto` ... getChild(0) returns a raw pointer and unless you have a reference somewhere else, your call to removeChild will destroy it. Use a ref_ptr instead.

**Reply by Mattc2021** [reply]
Ah you're right, I'm not incorporating the GeoPoint yet, right now I'm trying to figure out if a "look-to-point" type function is going to work. Right now I'm just trying to get the model to rotate a certain amount based off a column that is clicked on in an ImGui Table.

**Reply by gwaldron** [reply]
There is also a Matrix function that takes two osg::Vec3d's - it rotates one vector onto another - that may be handy for making a look-to-point matrix.
